lib/most_frequent_integer.rb

```diff
@@ -3,5 +3,8 @@
 # exists, return nil.
 
 def find_most_frequent_integer(arr)
-  # put your code here
+    num_val = Hash.new(0)                   #This is the new empty hash
+    arr.each {|arr| num_val[arr] += 1}      #Here we iterate over the array, puting it into a hash with the key being the number and the value how many
+    num_val.key(num_val.values.max)         #Here we iterate over the hash and find the key with the highest value
 end
+find_most_frequent_integer(arr)
```

Comment:
You should remove this line. It's causing an error that's preventing `rake` from running, so you aren't able to run the tests.

I think your code still needs a small change after that, but it's definitely close.